app/libs/rosa_clusters.py

```diff
@@ -0,0 +1,316 @@
+    ocm_env_url = (
+        None if ocm_env == "production" else f"https://api.{ocm_env}.openshift.com"
+    )
+    return ocm_token, ocm_env, ocm_env_url
+
+
+def get_cluster_object(ocm_token, ocm_env, cluster_data):
+    ocm_client = get_ocm_client(ocm_token, ocm_env)
+    for sample in TimeoutSampler(
+        wait_timeout=tts(ts="5m"),
+        sleep=1,
+        func=Cluster,
+        client=ocm_client,
+        name=cluster_data["cluster-name"],
+    ):
+        if sample and sample.exists:
+            return sample
+
+
+def prepare_managed_clusters_data(clusters, ocm_token, ocm_env):
+    for _cluster in clusters:
+        _cluster["cluster-name"] = _cluster["name"]
+        _cluster["ocm-token"] = ocm_token
+        _cluster["ocm-env"] = ocm_env
+        _cluster["timeout"] = tts(ts=_cluster.get("timeout", "30m"))
+        if _cluster["platform"] == HYPERSHIFT_STR:
+            _cluster["hosted-cp"] = "true"
+            _cluster["tags"] = "dns:external"
+            _cluster["machine-cidr"] = "10.0.0.0/16"
+
+        expiration_time = _cluster.get("expiration-time")
+        if expiration_time:
+            _cluster[
+                "expiration-time"
+            ] = f"{(datetime.now() + timedelta(hours=expiration_time)).isoformat()}Z"
+
+    return clusters
+
+
+def rosa_create_cluster(cluster_data):
+    hosted_cp_arg = "--hosted-cp"
+    ignore_keys = (
+        "name",
+        "platform",
+        "ocm-env",
+        "ocm-token",
+        "install-dir",
+        "timeout",
+        "auth-dir",
+    )
+    ocm_token, ocm_env, ocm_env_url = extract_ocm_data_from_cluster_data(cluster_data)
+    command = "create cluster --sts "
+    command_kwargs = {
+        f"--{_key}={_val}"
+        for _key, _val in cluster_data.items()
+        if _key not in ignore_keys
+    }
+
+    for cmd in command_kwargs:
+        if hosted_cp_arg in cmd:
+            command += f"{hosted_cp_arg} "
+        else:
+            command += f"{cmd} "
+
+    if cluster_data["platform"] == HYPERSHIFT_STR:
+        cluster_data = create_oidc(cluster_data=cluster_data)
+        cluster_data = prepare_hypershift_vpc(cluster_data=cluster_data)
+
+    dump_cluster_data_to_file(cluster_data=cluster_data)
+
+    rosa.cli.execute(
+        command=command,
+        ocm_env=ocm_env_url,
+        token=ocm_token,
+        aws_region=cluster_data["region"],
+    )
+
+    cluster_object = get_cluster_object(
+        ocm_token=ocm_token, ocm_env=ocm_env, cluster_data=cluster_data
+    )
+    cluster_object.wait_for_cluster_ready(wait_timeout=cluster_data["timeout"])
+
+    auth_path = os.path.join(cluster_data["install-dir"], "auth")
+    Path(auth_path).mkdir(parents=True, exist_ok=True)
+
+    with open(os.path.join(auth_path, "kubeconfig"), "w") as fd:
+        fd.write(yaml.dump(cluster_object.kubeconfig))
+
+    with open(os.path.join(auth_path, "kubeadmin-password"), "w") as fd:
+        fd.write(cluster_object.kubeadmin_password)
```

Comment:
different func?

Reply:
why?

Reply:
to have one place to do it and make it easier to follow the function

Reply:
Done